[WIP] v6 Docs Dev (jschalz) (#2025)
loft-sh/devspace: docs/pages/guides/file-synchronization.mdx

```

### Ignore `.git/`

> The sync can fail when files are constantly being changed while they are being synchronized. DevSpace will retry failing sync attempts but for folder such as `.git/` which contain continuously changing information and files which may be locked by the IDE, we recommmend to [ignore them via the `excludePaths` option](../configuration/dev/files/sync.mdx#exclude-paths) in `devspace.yaml`.

Reviewer: ```suggestion
The sync can fail when files are constantly being changed while they are being synchronized. DevSpace will retry failing sync attempts, but for a folder such as `.git/`, which contains continuously changing information and files which may be locked by the IDE, we recommend you [ignore them via the `excludePaths` option](../configuration/dev/files/sync.mdx#exclude-paths) in `devspace.yaml`.
```

### File Permissions

> Without file write permission, the sync will not be able to work. If you start your containers (in production) using a different user than `root` and this user does not have sufficient permissions to read and write certain files, you can:

Reviewer: ```suggestion
Without file write permission, the sync will not be able to work. If you start your containers (in production) using a different user than `root`, and this user does not have sufficient permissions to read and write certain files, you can:
```

- exclude these files using the [`excludePaths` option](../configuration/dev/files/sync.mdx#exclude-paths)
- exclude these files using the [`uploadExcludePaths` option](../configuration/dev/files/sync.mdx#example-exclude-paths-from-synchronization)
- run the containers using a different user during development\*

\* You can achieve this by:

> - Adding a `USER` statement in your Dockerfile: This is especially recommended when you are using multi-stage builds because you can add the `USER` statement for your development/build stage and add another `USER` statement to your production stage. Then, control the build target using the [`options.target` option](../configuration/images/build-settings/target.mdx) for Docker and kaniko builds and define a profile for removing the build stage for production deployments.

Reviewer: ```suggestion
- Adding a `USER` statement in your Dockerfile: This is especially recommended when using multi-stage builds, since you can add the `USER` statement for your development/build stage and add another `USER` statement to your production stage. Then, you can control the build target using the [`options.target` option](../configuration/images/build-settings/target.mdx) for Docker and kaniko builds, and define a profile for removing the build stage for production deployments.
```
